### PR TITLE
git-secrets-check PATH update

### LINKED ISSUE #1406: git-secrets is not run
**Description:**
Current runs show the following during the script runtime:
```git: 'secrets' is not a git command. See 'git --help'```

We need to correct this as a direct call to /tmp/git-secrets/git-secrets doesn't appear to be working and the local shell is trying to call the `git` binary in it's path instead.

Example of the failure [here](https://app.circleci.com/pipelines/github/raft-tech/TANF-app/5682/workflows/48e62539-333e-40d2-b1ed-b807f5779410/jobs/12619/parallel-runs/0/steps/0-102).

**Acceptance Criteria:** 
- [ ] git-secrets is actually run in pipelines
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
- [ ] Correct pathing so git-secrets is called correctly
- [ ] Add check for failure of previous command (e.g., `if [ $? -ne 0 ]` or `set -e`)
- [ ] Run Testing Checklist and confirm all tests pass

**Notes:**

**Supporting Documentation:**

**Open Questions:**

### PR DESCRIPTION
**Pull request by abottoms-coder**
### Summary of Changes
closes #1406 

**Acceptance Criteria:** 
- [x] git-secrets is actually run in pipelines
- [x] Testing Checklist has been run and all tests pass
- [x] reviewed and tested by @ADPennington 

### How to Test
1. Pull down CircleCI docker image and shell into it:
```
sudo docker run --rm -it --entrypoint bash cimg/python:3.8.10
```

1. Clone git-secrets manually and update config:
```
$ git clone https://github.com/awslabs/git-secrets.git # you must use https, github errors using ssh URI
$ vi .git/config
~~
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
[remote "origin"]
        url = https://github.com/awslabs/git-secrets.git
        fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
        remote = origin
        merge = refs/heads/master
[secrets]
        providers = git secrets --aws-provider
        patterns = (A3T[A-Z0-9]|AKIA|AGPA|AIDA|AROA|AIPA|ANPA|ANVA|ASIA)[A-Z0-9]{16}
        patterns = (\"|')?(AWS|aws|Aws)?_?(SECRET|secret|Secret)?_?(ACCESS|access|Access)?_?(KEY|key|Key)(\"|')?\\s*(:|=>|=)\\s*(\"|')?[A-Za-z0-9/\\+=]{40}(\"|')?
        patterns = (\"|')?(AWS|aws|Aws)?_?(ACCOUNT|account|Account)_?(ID|id|Id)?(\"|')?\\s*(:|=>|=)\\s*(\"|')?[0-9]{4}\\-?[0-9]{4}\\-?[0-9]{4}(\"|')?
        patterns = .+_KEY=.+
        allowed = [A-Z]+_KEY=..echo \".{S3_CREDENTIALS}\" [|] jq -r .+
        allowed = ./tdrs-backend/.env.example:.*
        allowed = ./tdrs-backend/docker-compose.yml:57:.*
        allowed = regexes.json:.*
        allowed = ./scripts/copy-login-gov-keypair.sh:14:JWT_KEY=.*
        allowed = ./scripts/set-backend-env-vars.sh:24:DJANGO_SECRET_KEY=.*
        allowed = .git/config:.*
        allowed = .gitconfig:.*

```
2. Create `test.sh`:
```
$vi test.sh
~~~
#!/bin/bash
git clone https://github.com/awslabs/git-secrets.git /tmp/git-secrets/

bash /tmp/git-secrets --scan -r
echo "exit code: $?"
```
3. See message `git 'secrets' is not a command` message is present.
4. Add path extension:
``` 
$vi test.sh
~~~
#!/bin/bash
git clone https://github.com/awslabs/git-secrets.git /tmp/git-secrets/

PATH="$PATH:/tmp/git-secrets"
bash /tmp/git-secrets --scan -r
echo "exit code: $?"
```
5. Verify message  `git 'secrets' is not a command` is not present.

> *Demo GIF(s) and screenshots for testing procedure*
image

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

+ [x] Look up the acceptance criteria in the related issue; paste ACs below in checklist format.
+ [ ] Check against the criteria:

As Product Owner, @lfrohlich will decide if ACs are met.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)
  + [x] Backend coverage: ___ (see https://github.com/raft-tech/TANF-app README coverage dashboard)

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [x] Does this PR change any linting or CI settings?

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*
+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with @iamjolly and @ttran-hub using Accessibility Insights reveal any errors introduced in this PR?
    + [See the full Accessibility Assessment plan here.](https://github.com/HHS/TANF-app/blob/main/docs/Technical-Documentation/how-government-will-test-a11y.md)

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful deployment by assigning a 'Deploy with CircleCI - {env}' label*

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

+ [ ] If this PR introduces backend code, is that code documented both inline and overall?
+ [ ] If this PR introduces frontend code, is that code documented both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

+ [x] Does the OWASP Scan pass on CircleCI?
+ [x] Do manual code review and manual testing detect any security issues?

### Deliverable 8: Context

> * Performance Standard(s): Code must be understandable and contextualized for the reviewers possess the knowledge and background necessary for analysis and constructive criticism to take place.

> * Acceptable Quality Level: Code submitted in the pull request has context.*
+ [x] Does this pull request contain sufficient inline comments providing relevant context for code snippets?
+ [x] Is the code understandable and lucid?
+ [x] Does this pull request provide background for why coding decisions were made?
+ [x] Can you as a reviewer explain and take ownership of these elements presented in this code review?

### CODE REVIEW

File: .github/pull_request_template.md
Comment (abottoms-coder) [inline .github/pull_request_template.md:3]: Because this is so quick, I didn't want to create another full review for it. Needed to remove "issue" word and wanted to keep it a full sentence.

File: scripts/git-secrets-check.sh
Comment (abottoms-coder) [inline scripts/git-secrets-check.sh:23]: Capture return code in variable for storage and use in L26.

File: scripts/git-secrets-check.sh
Comment (abottoms-coder) [inline scripts/git-secrets-check.sh:20]: If we can invoke it as `git secrets` vs `/tmp/git-secrets`, that would be preferable. Perhaps the `--install` arg can make this possible and eliminate the need for this PATH hijack.

**Comment by abottoms-coder** [comment]
Frontend is now failing due to Cypress tests. Cannot merge until #1427 is merged.

**Review by jtwillis92 — APPROVED** [review]
LGTM, I pulled the script down locally and `git secrets` is now available on my PATH as expected. Looks like there are some pre-existing relative path issues where it doesn't run all the way locally, but that may be a separate issue:
```
❯ bash -x ./scripts/git-secrets-check.sh                                                                                                                                                                                                                              ─╯
+ set -e
+ '[' -d /tmp/git-secrets ']'
+ echo The command git-secrets is available
The command git-secrets is available
+ '[' -f ../.gitconfig ']'
```

**Review by riatzukiza — APPROVED** [review]
LGTM

**Review by ADPennington — APPROVED** [review]
lgtm 👍🏾 approved pending ci steps pass.

**Comment by valcollignon** [comment]
demo by @abottoms-coder 12.7.21

**Pull request merged**
